tests/conftest.py

```diff
@@ -68,6 +68,12 @@ def pytest_sessionstart(session):
     cache.set("segfault", True)
 
 
+def pytest_runtest_setup(item):
+    if sys.implementation.name == "pypy":
+        if any(mark.name == "timing" for mark in item.iter_markers()):
+            pytest.skip("timing tests are unstable")
```

Comment:
I skipped all of the timing tests. Initially I tried to come up with a different tolerance, but it was unstable.

Reply:
The test marks are meant to control the test suite from the outside. You can add `timing` to the `NOT_MARKERS` env var.

```yaml
      - name: Exclude certain tests from pypy
        if: ${{ matrix.ext == 'pypy' }}  # figure out the check
        run: |
          echo "NOT_MARKERS=$NOT_MARKERS timing" >> $GITHUB_ENV
```